src/cider/nrepl/middleware/stacktrace.clj

```diff
@@ -68,10 +74,50 @@
   (let [tool-regex #"^clojure\.lang\.Compiler|^clojure\.tools\.nrepl|^cider\."
         tool? #(re-find tool-regex (or (:name %) ""))
         flag  #(if (tool? %)
-                 (update-in % [:flags] (comp set conj) :tooling)
+                 (flag-frame % :tooling)
                  %)]
     (map flag frames)))
 
+(def directory-namespaces
+  "This looks for all namespaces inside of directories on the class
+  path, ignoring jars."
+  (let [paths (-> (System/getProperty "java.class.path")
+                  (str/split #":"))]
+    (->> paths
+         (filter (complement #(.endsWith % ".jar")))
+         (map io/file)
+         (filter #(.isDirectory %))
+         (nsfind/find-namespaces)
+         (into #{}))))
+
+(def ns-common-prefix
+  "In order to match more namespaces, we look for a common namespace
+  prefix across the ones we have identified."
+  (let [common
+        (try (reduce
+              (fn [common ns]
+                (let [ns (str/lower-case ns)
+                      matched (map vector common ns)
+                      coincident (take-while (fn [[a b]] (= a b)) matched)]
+                  (apply str (map first coincident))))
+              (str/lower-case (first directory-namespaces))
+              directory-namespaces)
+             (catch Throwable e "NO_PREFIX_INFO"))]
```

Comment:
I think I'm getting a glimmer of an idea. I'll do it after work and make the data a little more explicit to follow more idiomatic Clojure. Good catch. Always appreciate your suggestions.